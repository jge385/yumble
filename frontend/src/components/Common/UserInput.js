import React from 'react';
import './UserInput.css';

/**
 * @param {*} props
 * @return {*}
 * 
 * This is the component to take user inputs.
 * Acts as a textfield for users to inputs such as Name and Group Name.
 * 
 */
function UserInput(props) {
  let classname;

  props.inputType == 'preferences' ?
    (classname = 'Preferences') :
    (classname = 'JoinGroup');

  return (
    <>
      <input
        className={classname}
        type={props.type}
        placeholder={props.placeholder}
        fontSize={props.fontSize}
        onChange={props.onChange}
      />
    </>
  );
}

export default UserInput;
