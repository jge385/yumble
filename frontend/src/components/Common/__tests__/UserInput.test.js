import React from 'react';
import renderer from 'react-test-renderer';
import UserInput from '../UserInput';

it('User Input snapshot testing', () => {
  const component1 = renderer.create(
      <>
        <UserInput inputType={'preferences'} />
      </>,
  );

  const tree1 = component1.toJSON();
  expect(tree1).toMatchSnapshot();

  const component2 = renderer.create(
      <>
        <UserInput inputType={'JoinGroup'} />
      </>,
  );
  const tree2 = component2.toJSON();
  expect(tree2).toMatchSnapshot();
});
