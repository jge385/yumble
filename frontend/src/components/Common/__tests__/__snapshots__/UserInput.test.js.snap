// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Input snapshot testing 1`] = `
<input
  className="Preferences"
/>
`;

exports[`User Input snapshot testing 2`] = `
<input
  className="JoinGroup"
/>
`;
